LimeSurvey/react-app
# Qe 119 add storybook tests for the text question type

1. Added storybook Jest addon for interaction tests.
2. Added the following tests for the ShortText Question
    1. Expect the question to have only one answer field
    2. Expect the question answer tag to be of type INPUT
    3. Expect the input field to accept numbers and letters
    4. Expect the question input to accept no more than 5 chars if attribute max_chars is set to 5
    5. Expect the question input type to be of type number if attribute only_numbers is true
    6. Expect the question input to accept only numbers if attribute only_numbers is true
3. Added the following tests for the LongText Question
     1. Expect the question to have only one answer field
     2. Expect the question answer tag to be of type TEXTAREA
     3. Expect the answer textarea to have 4 rows
     4. Expect the input field to accept numbers and letters
     5. Expect the question input to accept no more than 5 chars if attribute max_chars is set to 5
 4. Added the following tests for the HugeText Question
     1. Expect the question to have only one answer field
     2. Expect the question answer tag to be of type TEXTAREA
     3. Expect the answer textarea to have 12 rows
     4. Expect the input field to accept numbers and letters
     5. Expect the question input to accept no more than 5 chars if attribute max_chars is set to 5


How to run the storybook tests

1. Make sure the storybook is running `npm run storybook`
2. Run the storybook test `npm run test-storybook` by default it will check for port `6006`
3. You can also watch for changes `npm run test-storybook -- --watch `


![storybook tests](https://github.com/LimeSurvey/react-app/assets/129734454/01f6f06a-2282-4514-a52e-63ca3d5bfc01)
